event_project_follower/models/event_event.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# (c) 2016 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models, api
+
+
+class EventEvent(models.Model):
+    _inherit = 'event.event'
+
+    @api.model
+    def create(self, vals):
+        event = super(EventEvent, self).create(vals)
+        if vals.get('project_id', False):
+            event._add_followers_from_event_project()
+        return event
+
+    @api.multi
+    def write(self, vals):
+        super(EventEvent, self).write(vals)
+        if vals.get('project_id', False):
+            self._add_followers_from_event_project()
+        return True
+
+    def _add_followers_from_event_project(self):
+        follower_obj = self.env['mail.followers']
+        for event in self:
+            for user in event.project_id.members:
```

Comment:
```
for partner in (event.project_id.mapped('members.partner_id').filtered(lambda x: x not in event.message_follower_ids) | event.project_id.mapped('follower_ids').filtered(lamba x: x not in event.message_follower_ids):
    follower_obj.create({'res_model': 'event.event',
                         'res_id': event.id,
                         'partner_id': partner.id}
```

De esa manera en un solo `for` haces todos, habría que comprobar exactamente que no lo duplique pero algo así es posible, no lo he validado, así que puede que algo este mal escrito.

Si no también ten en cuenta que con la nueva api no es necesario comparar por ids.